spec/http-gateway.did

```diff
@@ -0,0 +1,34 @@
+type HeaderField = record { text; text; };
+
+type HttpRequest = record {
+  method: text;
+  url: text;
+  headers: vec HeaderField;
+  body: blob;
+};
+
+type HttpResponse = record {
+  status_code: nat16;
+  headers: vec HeaderField;
+  body: blob;
+  upgrade : opt bool;
```

Comment:
Typo: `upgrade` -> `update`

Reply:
Unfortunately(?) this field _is_ called `upgrade`, as in “upgrade the call from query to an update call”. Very confusing… oh well.

See <https://github.com/dfinity/icx-proxy/pull/6/files#diff-42cb6807ad74b3e201c5a7ca98b911c5fa08380e942be6e4ac5807f8377f87fcR259>

Reply:
Probably to late to change, but would `promote` be a better name?

Reply:
Probably. Or `resend_as_update_call`. Or, in line with the pattern for the streaming callback, it could even pass a `func` ref.

In a way it's not too late: we can change this while keeping backward conpat in the implementations (there is only one so far) relatively easily (the proxy can simply keep both in it's internal expected type). I just wonder if we should do it with this PR, or maybe as a separate step, to get this in first.